src/main/java/seedu/address/model/ingredient/Ingredient.java

```diff
@@ -62,6 +77,83 @@ public boolean isSameIngredient(Ingredient otherIngredient) {
                 && otherIngredient.getAmount().equals(getAmount());
     }
 
+    /**
+     * Returns true if the ingredient is a solid ingredient.
+     * @return true if the ingredient is a solid ingredient, false otherwise.
+     */
+    public boolean isSolidIngredient() {
+        Set<String> ingredientNames = new HashSet<>();
+        ingredientNames.add(BOBA);
+        ingredientNames.add(PEARL);
+        ingredientNames.add(BROWN_SUGAR);
+        String ingredientName = this.getIngredientNameAsString();
+
+        if (ingredientNames.contains(ingredientName)) {
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Returns true if the ingerdient is in shortage, which is when their levels fall below the default level.
+     * @return true if the ingerdient is in shortage, false otherwise.
+     */
+    public boolean isIngredientInShortage() {
+        if (this.isSolidIngredient()) {
+            return isSolidIngredientInShortage();
+        } else {
+            return isLiquidIngredientInShortage();
+        }
+    }
+
+    /**
+     * Returns true if the solid ingerdient is in shortage.
+     * @return true if the solid ingerdient is in shortage, false otherwise.
```

Comment:
There is a spelling error in "ingredient"

Reply:
Noted, thanks!